clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -836,6 +836,23 @@ static llvm::Constant *getPrologueSignature(CodeGenModule &CGM,
   return CGM.getTargetCodeGenInfo().getUBSanFunctionSignature(CGM);
 }
 
+/// Function checks whether given DeclContext contains a topmost
+/// namespace with name "sycl"
+static bool checkIfDeclaredInSYCLNamespace(const Decl *D) {
+  const DeclContext *DC = D->getDeclContext()->getEnclosingNamespaceContext();
+  const auto *ND = dyn_cast<NamespaceDecl>(DC);
```

Comment:
Can this just be a cast?

Reply:
There are plenty of different Derived classes.
https://clang.llvm.org/doxygen/classclang_1_1DeclContext.html